UPGRADE-GUIDE.md

```diff
@@ -0,0 +1,465 @@
+# Ganache Upgrade Guide
+
+## ganache-core v2.x.x and ganache-cli v6.x.x to ganache v7.x.x
+
+### Installing
+
+The Ganache npm packages have been combined and renamed from `ganache-cli` and
+`ganache-core` to `ganache`. You may need to uninstall the old packages before
+installing the new ganache.
+
+There are two ways you may have the old packages installed, globally or locally.
+The uninstall and installation process are slightly different for each.
+
+#### Globally installed
+
+A global installation makes it possible to run Ganache from your command line at
+any working directory: `$~ ganache [...options]`.
+
+To upgrade a global installation using [`npm`](____) run these two commands:
+
+```console
+$ npm uninstall --global ganache-cli ganache-core
+$ npm install --global ganache
+```
+
+_For security reasons[^1] we do not recommend using yarn or pnpm as your package
+manager when installing security-critical applications like Ganache._
+
+You will now be able to run `$~ ganache` from your command line.
+
+_note: you may need to uninstall using yarn, or another node package manager,
+depending on how you originally installed ganache-cli or ganache-core._
+
+#### Locally installed
+
+A local installation makes it possible to `import` or `require` Ganache from 
+programmatically from JavaScript or TypeScript. Additionally, you can run the
+command line version of Ganache from your [package.json scripts](https://docs.npmjs.com/cli/v8/using-npm/scripts) or directly
+from the command line using `npx ganache` ([what is npx?](https://www.npmjs.com/package/npx)).
+
+To upgrade a local installation using `npm` run these two commands:
+
+```console
+$ npm uninstall ganache-cli ganache-core
+$ npm install --save-dev ganache
+```
+
+_For security reasons[^1] we do not recommend using yarn or pnpm as your package
+manager when installing security-critical applications like Ganache._
+
+You will now be able to run Ganache programmatically from within your JavaScript
+or TypeScript files.
+
+An example diff might look like:
+
+```diff
+diff --git a/index.js b/index.js
```

Comment:
For me, the first 4 lines of the diff seemed unnecessary and led to a few seconds of confusion. Probably just due to inexperience with console diff tools, but I don't really see the value these lines are adding.

Reply:
section removed